### DISCUSSION TITLE
ValueError: The size of the proposed random crop ROI is larger than the image size.

### DISCUSSION THREAD

**Discussion by dannyhow12** [Project-MONAI/tutorials#706] [Q&A] [answered]
Hi everyone! I am working with the KiTS21 dataset, and is now just testing another method by using the MONAI pipeline. Only 10 cases of kidney CT scan images are used, where it includes the raw images as well as the segmentation masks. It was working just fine yesterday when I directly implemented the spleen 3D segmentation .ipynb code for it. 

I attempted to adjust the parameters to my own application, setting the below: 

    train_transforms = Compose(
        [
            LoadImaged(keys=["image", "label"]), #Loading NIFTI Images
            EnsureChannelFirstd(keys=["image", "label"]),
            Orientationd(keys=["image", "label"], axcodes="RAS"), #Right anterior superior // (L,R); (A,P); (I,S)
            Spacingd(keys=["image", "label"], pixdim=(
                1.62, 1.62, 3.22), mode=("bilinear", "nearest")),
            ScaleIntensityRanged(
                keys=["image"], a_min=-79, a_max=304,
                b_min=0.0, b_max=1.0, clip=True,
            ), #Clip = whether to clip after scaling is performed #-79 304 for kits
            CropForegroundd(keys=["image", "label"], source_key="image"),
            RandCropByPosNegLabeld(
                keys=["image", "label"],
                label_key="label",
                spatial_size=(128, 128, 64), #default 96,96,96
                pos=1,
                neg=1, # pos / (pos+neg) = probability to pick fg than bg
                num_samples=4, # no. of crop regions to take in each list
                image_key="image",
                image_threshold=0,
            ),
            # user can also add other random transforms
            # RandAffined(
            #     keys=['image', 'label'],
            #     mode=('bilinear', 'nearest'),
            #     prob=1.0, spatial_size=(96, 96, 96),
            #     rotate_range=(0, 0, np.pi/15),
            #     scale_range=(0.1, 0.1, 0.1)),
            EnsureTyped(keys=["image", "label"]),
        ]
    )
    val_transforms = Compose(
        [
            LoadImaged(keys=["image", "label"]),
            EnsureChannelFirstd(keys=["image", "label"]),
            Orientationd(keys=["image", "label"], axcodes="RAS"),
            Spacingd(keys=["image", "label"], pixdim=(
                1.62, 1.62, 3.22), mode=("bilinear", "nearest")),
            ScaleIntensityRanged(
                keys=["image"], a_min=-79, a_max=304,
                b_min=0.0, b_max=1.0, clip=True,
            ),
            CropForegroundd(keys=["image", "label"], source_key="image"),
            EnsureTyped(keys=["image", "label"]),
        ]
    )

1) Prior to that, I have also set my train_files and val_files to a ratio of 6:4, with 6 training cases and 4 validation cases.
2) Number of workers was set to 2 as there was a suggestion that the max number of workers should be 2 instead of 4
3) Batch size for val_loader was set to 2 instead of 1, as I wanted to synchronise it with train_loader.

Take note on 3), where I believe is the source of the issue. Can anyone explain why? 

https://colab.research.google.com/drive/1C0SKDczZGu6JwXX8lZHx5H6tmczRcd5Z#scrollTo=wYc5d-w6YttR This is my code if anyone would want to view it. 

**Note**
It is now working if I set the batch_size for val_loader to be 1. In this new implementation, I have increased the number of datasets that I am using, which is 100. After running all the blocks of the code until when I need to initiate the typical training process, this error pops up after training for several iterations. 

ValueError: The size of the proposed random crop ROI is larger than the image size.

My original image size is 512 x 512 x A number of slices, while the proposed spatial size in RandCropByPosNegLabeld is 128,128,64. I am unsure why this error pops up

**Comment by Nic-Ma** [comment, marked as answer]
Hi @dannyhow12 ,

Thanks for your interest and experiments here.
I think the root cause is that: your crop ROI is `spatial_size=(128, 128, 64)`, but some of your images after `spacing` transform is smaller than the ROI size. Could you please help pad the image or reduce the ROI?

Thanks.

**Reply by ferchonavarro** [reply]
Could it be possible to automatically pad the image if the size doesn't match? Otherwise, how can you use this with images that don't
meet the cropping criteria?

**Reply by Ontheroad123** [reply]
https://github.com/Project-MONAI/MONAI/discussions/6164#discussioncomment-5413284，i solve this error by automatically pad the image if the size doesn't match, this may help you.
